core/src/main/java/org/infinispan/functional/impl/Params.java

```diff
@@ -108,4 +115,50 @@ public static Params from(Param<?>... ps) {
       }
    }
 
+   public static class Externalizer implements AdvancedExternalizer<Params> {
+      @Override
+      public Set<Class<? extends Params>> getTypeClasses() {
+         return Collections.singleton(Params.class);
+      }
+
+      @Override
+      public Integer getId() {
+         return Ids.PARAMS;
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, Params object) throws IOException {
+         for (Param param : object.params) {
+            switch (param.serializeMethod()) {
```

Comment:
I see what you are doing here but this is brittle IMO. Simply have each Param have its own Externalizer and that should work without any further complications nor the need to add an extra method to the interface. Besides, I don't expect that many params to make this is really that useful. IOW, I expect the number of Param impls to be in the tens and not in the hundreds :)